lib/cockpit-components-select.jsx

```diff
@@ -39,16 +39,24 @@ define([
             onChange: React.PropTypes.func,
             id: React.PropTypes.string,
         },
+        handleDocumentClick: function(ev) {
+            // clicking outside the select control should ensure it's closed
+            if (!React.findDOMNode(this).contains(ev.target))
```

Comment:
> You know React better than I. With my limited knowledge I would be wary that it somehow returns different results after the document has been mutated ...

This is guarded against by componentDidMount and componentWillUnmount - the function will only be called for an instance of this class mounted in the current DOM.